### PR TITLE
Add new methods markCollectionsAsDeleted and markWorkbooksAsDeleted

### PR DESCRIPTION
**Pull request by Sergey-weber**

### CODE REVIEW

File: src/registry/common/entities/collection/collection.ts [hunk 1]
```diff
@@ -88,6 +88,10 @@ export const Collection: CollectionConstructor = class Collection implements Col
         this.permissions = permissions;
     }
 
+    async deletePermissions(): Promise<void> {
+        this.permissions = undefined;
```
Comment (imsitnikov) [inline src/registry/common/entities/collection/collection.ts:92]: There is no need to clear `this.permissions` here. These methods only for clearing permissions in external services.

File: src/services/new/collection/delete-collections.ts [hunk 2]
```diff
@@ -48,66 +45,66 @@ export const deleteCollections = async (
...
+        .select()
+        .from(recursiveName)
+        .timeout(CollectionModel.DEFAULT_QUERY_TIMEOUT);
+
+    const collectionsForDeleteIds = collectionsForDelete.map((item) => item.collectionId);
+
+    const collectionsMap = await getCollectionsListByIds(
         {ctx, trx: targetTrx, skipValidation, skipCheckPermissions},
-        {collectionIds, permission: CollectionPermission.Delete},
+        {collectionIds: collectionsForDeleteIds, permission: CollectionPermission.Delete},
```
Comment (imsitnikov) [inline src/services/new/collection/delete-collections.ts:86]: We start checking the `delete` permission for every collection in the tree, this is wrong, we only need to check direct deleting collections.

File: src/services/new/collection/delete-collections.ts [hunk 3]
```diff
@@ -48,66 +45,66 @@ export const deleteCollections = async (
...
+    const workbooksForDelete = await WorkbookModel.query(getReplica(trx))
+        .select()
+        .where(WorkbookModelColumn.CollectionId, 'in', collectionsForDeleteIds)
+        .where({
+            [WorkbookModelColumn.DeletedAt]: null,
+        })
+        .timeout(WorkbookModel.DEFAULT_QUERY_TIMEOUT);
+
+    for (const workbook of workbooksForDelete) {
+        if (workbook.isTemplate) {
```
Comment (imsitnikov) [inline src/services/new/collection/delete-collections.ts:98]: Looks like this check already exists in the `deleteWorkbooks` method.

**Review by imsitnikov — CHANGES_REQUESTED** [review]

File: src/services/new/collection/delete-collections.ts (hunk 2 above)
Comment (Sergey-weber) [reply src/services/new/collection/delete-collections.ts]: yes, fixed it

File: src/services/new/collection/delete-collections.ts (hunk 3 above)
Comment (Sergey-weber) [reply src/services/new/collection/delete-collections.ts:127]: yes, but there is a more informative error - "Collection with workbook template can't be deleted"? and if we fall here, will we not have to request workbooks once again.  this code was already there, I just moved it

**Review by imsitnikov — CHANGES_REQUESTED** [review]

**Review by imsitnikov — APPROVED** [review]

**Pull request merged**